src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -5116,6 +5140,246 @@ static synchronized long getColumnEncryptionKeyCacheTtl() {
         return columnEncryptionKeyCacheTtl;
     }
 
+    
+    /**
+     * Used to keep track of an individual handle ready for un-prepare.
+     */
+    private final class PreparedStatementDiscardItem  {
+
+        public int handle; 
+        public boolean directSql;
+
+        public PreparedStatementDiscardItem(int handle, boolean directSql) {
+            this.handle = handle;
+            this.directSql = directSql;
+        }
+    }
+
+
+    /**
+     * Enqueue a discarded prepared statement handle to be clean-up on the server.
+     * 
+     * @param handle
+     *      The prepared statement handle
+     * @param directSql
+     *      Whether the statement handle is direct SQL (true) or a cursor (false)
+     */
+    final void enqueuePreparedStatementDiscardItem(int handle, boolean directSql)
+    {
+        if (getConnectionLogger().isLoggable(java.util.logging.Level.FINER))
+            getConnectionLogger().finer(this + ": Adding PreparedHandle to queue for un-prepare:" + handle);
+
+        // Add the new handle to the discarding queue and find out current # enqueued.
+        this.discardedPreparedStatementHandles.add(new PreparedStatementDiscardItem(handle, directSql));
+        this.discardedPreparedStatementHandleQueueCount.incrementAndGet();
+    }
+
+
+    /**
+     * Returns the number of currently outstanding prepared statement un-prepare actions.
+     */
+    public int outstandingPreparedStatementDiscardActionCount(){
+        return this.discardedPreparedStatementHandleQueueCount.get();
+    }
+
+    /**
+     * Forces the un-prepare requests for any outstanding discarded prepared statements to be executed.
+     */
+    public void forcePreparedStatementDiscardActions()
```

Comment:
There a bunch of formatting issues. Can you please use an IDE to format this. Ctrl A and Crtl Shift F in Eclipse should take care of this.